$NetBSD: patch-Makefile,v 1.4 2018/01/10 09:36:54 adam Exp $

Remove -O3 -g compiler flags.
Disable -march=native since it causes problems on NetBSD.
Use the same shared library name across platforms.

--- Makefile.orig	2017-12-27 19:56:31.000000000 +0000
+++ Makefile
@@ -30,7 +30,7 @@ SRC_BENCH = src/bench.c
 SRC_GENKAT = src/genkat.c
 OBJ = $(SRC:.c=.o)
 
-CFLAGS += -std=c89 -O3 -Wall -g -Iinclude -Isrc
+CFLAGS += -std=c89 -Wall -Iinclude -Isrc
 
 ifeq ($(NO_THREADS), 1)
 CFLAGS += -DARGON2_NO_THREADS
@@ -50,7 +50,6 @@ $(info Building without optimizations)
 	SRC += src/ref.c
 else
 $(info Building with optimizations for $(OPTTARGET))
-	CFLAGS += -march=$(OPTTARGET)
 	SRC += src/opt.c
 endif
 
@@ -59,7 +58,7 @@ KERNEL_NAME := $(shell uname -s)
 
 LIB_NAME=argon2
 ifeq ($(KERNEL_NAME), Linux)
-	LIB_EXT := so.$(ABI_VERSION)
+	LIB_EXT := so
 	LIB_CFLAGS := -shared -fPIC -fvisibility=hidden -DA2_VISCTL=1
 	SO_LDFLAGS := -Wl,-soname,lib$(LIB_NAME).$(LIB_EXT)
 	LINKED_LIB_EXT := so
@@ -69,8 +68,8 @@ ifeq ($(KERNEL_NAME), $(filter $(KERNEL_
 	LIB_CFLAGS := -shared -fPIC
 endif
 ifeq ($(KERNEL_NAME), Darwin)
-	LIB_EXT := $(ABI_VERSION).dylib
-	LIB_CFLAGS := -dynamiclib -install_name @rpath/lib$(LIB_NAME).$(LIB_EXT)
+	LIB_EXT := dylib
+	LIB_CFLAGS := -dynamiclib -install_name ${DESTDIR}${PREFIX}/lib/lib$(LIB_NAME).$(LIB_EXT)
 	LINKED_LIB_EXT := dylib
 endif
 ifeq ($(findstring CYGWIN, $(KERNEL_NAME)), CYGWIN)
